### PR TITLE
RHDEVDOCS-4124 Operator displays Trigger metrics in devconsole

### PR DESCRIPTION
**Pull request by sounix000**
- **Aligned team**: Dev Tools
- **OCP version for cherry-picking**: `enterprise-4.9`, `enterprise-4.10`, `enterprise-4.11`
- **JIRA issues**: [RHDEVDOCS-4124 Operator displays Trigger metrics in devconsole](https://issues.redhat.com/browse/RHDEVDOCS-4124)
- **Preview pages**: Due to issues with Netlify, preview pages are not getting generated. However, you can download the [locally generated HTML file](https://drive.google.com/file/d/1_Fc_57TFBRnufIed2c7DeJBHb70NYto5/view?usp=sharing) and open it in your browser. Look for the section "**Enabling monitoring of event listeners for Triggers for user-defined projects**".
- **SME review**: @savitaashture      
- **QE review**: @VeereshAradhya    
- **Peer-review**: @rolfedh

### CODE REVIEW

**Comment by savitaashture** [comment]
/lgtm

:bowing_woman:

File: cicd/pipelines/creating-applications-with-cicd-pipelines.adoc
Comment (rolfedh) [inline cicd/pipelines/creating-applications-with-cicd-pipelines.adoc:68]: From the checklist: “Additional resources” and “Next steps” titles use unique IDs."
How about:
```suggestion
[role="_additional-resources_op-enabling-monitoring-of-event-listeners-for-triggers-for-user-defined-projects"]
```

File: modules/op-enabling-monitoring-of-event-listeners-for-triggers-for-user-defined-projects.adoc
Comment (rolfedh) [inline modules/op-enabling-monitoring-of-event-listeners-for-triggers-for-user-defined-projects.adoc:9]: From the checklist: "Check for places where `OpenShift` should be `{product-title}`"
How about:
```suggestion
On receiving a HTTP request, event listeners for Triggers return three metrics -- `http_duration_seconds`, `event_count`, and `triggered_resources`. Cluster administrators can configure the monitoring of event listener metrics for Triggers in a user-defined project. When configured, users and administrators can view the metrics in the {product-title} web console.
```
or
```suggestion
On receiving a HTTP request, event listeners for Triggers return three metrics -- `http_duration_seconds`, `event_count`, and `triggered_resources`. Cluster administrators can configure the monitoring of event listener metrics for Triggers in a user-defined project. When configured, users and administrators can view the metrics in the web console.
```

File: modules/op-enabling-monitoring-of-event-listeners-for-triggers-for-user-defined-projects.adoc
Comment (rolfedh) [inline modules/op-enabling-monitoring-of-event-listeners-for-triggers-for-user-defined-projects.adoc:56]: Same here.

File: cicd/pipelines/creating-applications-with-cicd-pipelines.adoc
Comment (rolfedh) [inline cicd/pipelines/creating-applications-with-cicd-pipelines.adoc:72]: On line 2 of this file, even though it is outside the scope of this PR, please replace `[id='creating-applications-with-cicd-pipelines']` with `[id="creating-applications-with-cicd-pipelines"]`

File: modules/op-enabling-monitoring-of-event-listeners-for-triggers-for-user-defined-projects.adoc
Comment (rolfedh) [inline modules/op-enabling-monitoring-of-event-listeners-for-triggers-for-user-defined-projects.adoc:22]: Should they create exactly the service monitor shown, or is this just one possible example? If it is an example, how about saying:
```suggestion
. For each event listener, create a service monitor. For example, to view the metrics for the `github-listener` event listener in the `test` namespace, create a service monitor. For example:
+
[source,yaml]
```
OR
```suggestion
. For each event listener, create a service monitor. For example, to view the metrics for the `github-listener` event listener in the `test` namespace, create a service monitor.
+
.Example service monitor
[source,yaml]
```

**Review by rolfedh — COMMENTED** [review]
Nice work, @sounix000. I left some comments for your consideration.

File: modules/op-enabling-monitoring-of-event-listeners-for-triggers-for-user-defined-projects.adoc
Comment (rolfedh) [inline modules/op-enabling-monitoring-of-event-listeners-for-triggers-for-user-defined-projects.adoc:7]: @michaelryanpeter You mentioned using "Optional" in today's OpenShift docs team meeting. Do you have any suggestions or comments on this instance?
https://docs.google.com/document/d/1CH-K3UAuAGbeqp0W-9AHH2v7OrCQawsxt4fiDKLj86Q/edit?disco=AAAAakoYDio

Comment (rolfedh) [reply modules/op-enabling-monitoring-of-event-listeners-for-triggers-for-user-defined-projects.adoc:7]: - Using "optional" in titles is a complex issue that has come up for discussion recently.
- Red Hat seems to prefer not using Optional in titles this way. 
- If this optional module is the exception to a series of required tasks in an assembly, let's find a better way to make that obvious.  

```suggestion
= Enabling monitoring of event listeners for Triggers for user-defined projects
```

Comment (rolfedh) [reply modules/op-enabling-monitoring-of-event-listeners-for-triggers-for-user-defined-projects.adoc:7]:
```suggestion
As a cluster administrator, to gather event listener metrics for Triggers in a user-defined project and display them in the web console, you can create a service monitor for each event listener. On receiving a HTTP request, event listeners for Triggers return three metrics -- `http_duration_seconds`, `event_count`, and `triggered_resources`.
```

File: modules/op-enabling-monitoring-of-event-listeners-for-triggers-for-user-defined-projects.adoc
Comment (rolfedh) [reply modules/op-enabling-monitoring-of-event-listeners-for-triggers-for-user-defined-projects.adoc:9]:
```suggestion
As a cluster administrator, to gather event listener metrics for Triggers in a user-defined project and display them in the web console, you can create a service monitor for each event listener. On receiving a HTTP request, event listeners for Triggers return three metrics -- `http_duration_seconds`, `event_count`, and `triggered_resources`.
```

Comment (rolfedh) [reply modules/op-enabling-monitoring-of-event-listeners-for-triggers-for-user-defined-projects.adoc:9]: In response to removing "Optional:" from the title, please consider this ^^ alteration of the abstract, which puts the user story and condition first.

File: modules/op-enabling-monitoring-of-event-listeners-for-triggers-for-user-defined-projects.adoc
Comment (michaelryanpeter) [reply modules/op-enabling-monitoring-of-event-listeners-for-triggers-for-user-defined-projects.adoc:7]: The consensus from the OpenShift documentation team is that using "optional" in a heading creates problems for content reuse. As @rolfedh mentioned, explaining the conditional context of the procedure in the abstract is considered a best practice.

**Review by rolfedh — APPROVED** [review]
Great work! Thanks, Souvik! Please be sure to squash before requesting merge.

File: cicd/pipelines/creating-applications-with-cicd-pipelines.adoc
Comment (abrennan89) [inline cicd/pipelines/creating-applications-with-cicd-pipelines.adoc:69]: Missing an ID, e.g.
```
[role="_additional-resources"]
[id="additional-resources_creating-applications-with-cicd-pipelines"]
== Additional resources
```
please see https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#anchoring-prerequisites-additional-resources-and-next-steps-titles-in-assemblies

File: modules/op-enabling-monitoring-of-event-listeners-for-triggers-for-user-defined-projects.adoc
Comment (abrennan89) [inline modules/op-enabling-monitoring-of-event-listeners-for-triggers-for-user-defined-projects.adoc:10]: I think it should be "triggers", lowercase? 
see https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#api-object-formatting

File: modules/op-enabling-monitoring-of-event-listeners-for-triggers-for-user-defined-projects.adoc
Comment (abrennan89) [inline modules/op-enabling-monitoring-of-event-listeners-for-triggers-for-user-defined-projects.adoc:15]: I think web console should be a prerequisite here? e.g.
```
You have logged in to the {product-title} web console.
```

Comment (abrennan89) [inline modules/op-enabling-monitoring-of-event-listeners-for-triggers-for-user-defined-projects.adoc:15]:
```suggestion
* You have installed the {pipelines-title} Operator.
* You have enabled monitoring for user-defined projects.
```

File: modules/op-enabling-monitoring-of-event-listeners-for-triggers-for-user-defined-projects.adoc
Comment (abrennan89) [inline modules/op-enabling-monitoring-of-event-listeners-for-triggers-for-user-defined-projects.adoc:12]: Not sure why make these discrete headings? I don't think it's required.

**Review by abrennan89 — CHANGES_REQUESTED** [review]
Added some comments

File: cicd/pipelines/creating-applications-with-cicd-pipelines.adoc
Comment (rolfedh) [reply cicd/pipelines/creating-applications-with-cicd-pipelines.adoc:69]: Oops. My mistake. Thanks.

File: modules/op-enabling-monitoring-of-event-listeners-for-triggers-for-user-defined-projects.adoc
Comment (rolfedh) [reply modules/op-enabling-monitoring-of-event-listeners-for-triggers-for-user-defined-projects.adoc:10]: When I reviewed the `op*.adoc` files in openshift-docs, I noticed that "Triggers" was the dominant usage. It seems to be a short name for "Tekton Triggers," which is "a Kubernetes Custom Resource Definition (CRD) controller."

Comment (rolfedh) [reply modules/op-enabling-monitoring-of-event-listeners-for-triggers-for-user-defined-projects.adoc:10]: - The instances of "Triggers" in this document all seem to be proper nouns that reference this controller service, not `Trigger` resources. 
- Changing "Triggers" to lower-case might cause confusion. A reader might interpret "triggers" as something that triggers an event, not as the controller or resource.
Therefore, I recommend leaving "Triggers" unchanged.
@abrennan89 WDYT?

File: modules/op-enabling-monitoring-of-event-listeners-for-triggers-for-user-defined-projects.adoc
Comment (rolfedh) [reply modules/op-enabling-monitoring-of-event-listeners-for-triggers-for-user-defined-projects.adoc:12]: To learn more, I've searched openshift-docs and looked at:

- https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#discrete-headings
- https://docs.asciidoctor.org/asciidoc/latest/blocks/discrete-headings/

There are many instances of these `[discrete]` attributes above block headings (e.g., `.Prerequisites`) in openshift-docs. However, they seem unnecessary because asciidoc already excludes block headings from TOCs and such. 
It seems that this attribute only affects section titles (e.g., `== This is a section title`) that you want to make more discrete.

File: cicd/pipelines/creating-applications-with-cicd-pipelines.adoc
Comment (abrennan89) [inline cicd/pipelines/creating-applications-with-cicd-pipelines.adoc:70]: Honestly I'd get another pair of eyes from the core team to check this - I _think_ it needs to be a H2 instead of a .heading and the ID I believe should be per the assembly, not the module name before it? @rolfedh any idea?

Comment (abrennan89) [reply cicd/pipelines/creating-applications-with-cicd-pipelines.adoc:70]: tbh I think it would be better to just add this to the existing additional resources section: 
image

File: cicd/pipelines/creating-applications-with-cicd-pipelines.adoc
Comment (abrennan89) [inline cicd/pipelines/creating-applications-with-cicd-pipelines.adoc:69]:
```suggestion
[role="_additional-resources"]
```

**Review by abrennan89 — CHANGES_REQUESTED** [review]
one last change 🙏🏻

File: cicd/pipelines/creating-applications-with-cicd-pipelines.adoc
Comment (rolfedh) [reply cicd/pipelines/creating-applications-with-cicd-pipelines.adoc:70]: +1 I agree with @abrennan89' recommendation. Disregard my previous comment. 

Normally we would have moved the `.Additional resources` section like shown, because this is the recommended workaround for the "no xrefs in modules" issue. However, in this case, it looks wrong having two additional resources sections stacked on top of each other. I didn't notice that at the time I made the recommendation.

Comment (rolfedh) [reply cicd/pipelines/creating-applications-with-cicd-pipelines.adoc:70]: I support @abrennan89's recommendation. Let's eliminate the module-level `.Additional resources` and move the xref to the following `== Additional resources` section.

**Review by abrennan89 — APPROVED** [review]

File: modules/op-enabling-monitoring-of-event-listeners-for-triggers-for-user-defined-projects.adoc
Comment (VeereshAradhya) [inline modules/op-enabling-monitoring-of-event-listeners-for-triggers-for-user-defined-projects.adoc:9]: @savitaashture only the metrics name `triggered_resources` is enough or we need to add `eventlistener_triggered_resources`?

File: modules/op-enabling-monitoring-of-event-listeners-for-triggers-for-user-defined-projects.adoc
Comment (VeereshAradhya) [inline modules/op-enabling-monitoring-of-event-listeners-for-triggers-for-user-defined-projects.adoc:15]: @sounix000 Do we need to provide any link reference here?

Comment (sounix000) [reply modules/op-enabling-monitoring-of-event-listeners-for-triggers-for-user-defined-projects.adoc:15]: We cannot provide direct inline links due to limitations set by the larger OpenShift docs org. I have given one link in the **Additional resources** section immediately after the procedure, and some other relevant links are in the bigger Additional resources section at the bottom of the page.

File: modules/op-enabling-monitoring-of-event-listeners-for-triggers-for-user-defined-projects.adoc
Comment (sounix000) [reply modules/op-enabling-monitoring-of-event-listeners-for-triggers-for-user-defined-projects.adoc:9]: I also had the same doubt - there was a mismatch between the text and the screenshot in Savita's [draft document](https://docs.google.com/document/d/1uumFX8iHEGjSuPjMdNuB4x3ueyuXrOR2TKhLUTXo--0/edit#heading=h.n09pyxcib1k7).

Comment (savitaashture) [reply modules/op-enabling-monitoring-of-event-listeners-for-triggers-for-user-defined-projects.adoc:9]: Oh sorry yes its `eventlistener_triggered_resources` the one from screenshot 
For reference [follow](https://github.com/tektoncd/triggers/blob/main/docs/eventlisteners.md#configuring-metrics-for-eventlisteners)

Comment (VeereshAradhya) [reply modules/op-enabling-monitoring-of-event-listeners-for-triggers-for-user-defined-projects.adoc:9]: what about `http_duration_seconds` and `event_count`?

File: modules/op-enabling-monitoring-of-event-listeners-for-triggers-for-user-defined-projects.adoc
Comment (VeereshAradhya) [reply modules/op-enabling-monitoring-of-event-listeners-for-triggers-for-user-defined-projects.adoc:15]: Ack

File: modules/op-enabling-monitoring-of-event-listeners-for-triggers-for-user-defined-projects.adoc
Comment (savitaashture) [reply modules/op-enabling-monitoring-of-event-listeners-for-triggers-for-user-defined-projects.adoc:9]: yes for all `eventlistener_` is prefixed 
I have updated [draft doc](https://docs.google.com/document/d/1uumFX8iHEGjSuPjMdNuB4x3ueyuXrOR2TKhLUTXo--0/edit#) and also to follow from upstream doc here is the [reference](https://github.com/tektoncd/triggers/blob/main/docs/eventlisteners.md#configuring-metrics-for-eventlisteners)

**Review by VeereshAradhya — APPROVED** [review]

**Comment by abrennan89** [comment]
Merged per the peer review checklist: https://docs.google.com/document/d/1WN7k72PKxcPA__erp5TK-CefmS_sVwE3r40PvpmNk3E/edit#bookmark=id.htiwq68zv3ap

**Pull request merged**
